### PR TITLE
Button fixes

### LINKED ISSUE #608: Some buttons throw errors if no link is specified
CTA button will throw error if URL is empty. Right now we have mandatory links for CTA, but if code gets copied to other projects, it might be problematic.

### PR DESCRIPTION
**Pull request by balagan73**
#608 

- Handling empty CTA button
- External link of hero image opens in new tab

### CODE REVIEW

**Comment by balagan73** [comment]
image

**Review by bboro — APPROVED** [review]

**Comment by bboro** [comment]
@balagan73 test failing, merge when green

File: web/modules/custom/server_general/src/ElementTrait.php [hunk 1]
```diff
@@ -34,13 +35,13 @@ trait ElementTrait {
...
    *   The button text.
-   * @param \Drupal\Core\Url $url
+   * @param ?\Drupal\Core\Url $url
    *   The URL to link the button to.
    *
    * @return array
    *   Render array.
    */
-  protected function buildElementCta(string $title, array $body, string $button_text, Url $url): array {
+  protected function buildElementCta(string $title, array $body, string $button_text = '', Url $url = NULL): array {
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait.php:44]: Can we convert this to `protected function buildElementCta(string $title, array $body, Link $button_link): array {`

Also, URL should be required

Comment (balagan73) [reply web/modules/custom/server_general/src/ElementTrait.php:44]: I also prefer that, but I was afraid that wouldn't be backward compatible.

**Comment by balagan73** [comment]
Style guide|CTA paragraph
-|-
image|image

File: web/modules/custom/server_general/src/ElementTrait.php [hunk 2]
```diff
@@ -32,15 +33,14 @@ trait ElementTrait {
    *   The title.
    * @param array $body
    *   The body render array.
-   * @param string $button_text
-   *   The button text.
-   * @param \Drupal\Core\Url $url
-   *   The URL to link the button to.
+   * @param \Drupal\Core\Link $link
+   *   The button Link object.
+   *   If NULL, no button is rendered. Defaults to NULL.
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait.php:38]:
```suggestion
```

**Review by amitaibu — APPROVED** [review]
Thanks, MWG after fixes

**Pull request merged**